core/lib/multistore/index.ts

```diff
@@ -229,11 +210,12 @@ export function localizedRoute (routeObj: LocalizedRoute | string | RouteConfig
 
 export function setupMultistoreRoutes (config, router: VueRouter, routes: RouteConfig[], priority: number = 0): void {
   const allRoutes: RouteConfig[] = []
-  const { storeCode, appendStoreCode } = currentStoreView()
-  if (storeCode && appendStoreCode) {
+  const { storeCode } = currentStoreView()
+  if (config.defaultStoreCode !== storeCode && config.storeViews.multistore) {
     allRoutes.push(...routes.map(route => localizedRouteConfig(route, storeCode)))
   } else {
     allRoutes.push(...routes)
+    allRoutes.push(...routes.map(route => localizedRouteConfig(route, storeCode)))
```

Comment:
Shouldn't we prevent recurssion in `localizedRouteConfig` if `!storeCode`? Why do we have this map there?

Reply:
hmm `localizedRouteConfig` will just not add `url` or `storeCode` 🤔 but yeah it will create duplicated routes. I will change that